packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -316,6 +316,10 @@ export function createVM<HostNode, HostElement>(
         vm.toString = (): string => {
             return `[object:vm ${def.name} (${vm.idx})]`;
         };
+        // Forced mixed shadow mode for testing
+        if (process.env.FORCED_MIXED_SHADOW_MODE) {
+            vm.shadowMode = ShadowMode.Native;
+        }
```

Comment:
It would be nice if we could find a way to enable this in our tests without actually modifying the non-test code. I think as-is, this will end up in the built LWC files `packages/lwc/dist`, which means that consumers will have to replace the string (`process.env.FORCED_MIXED_SHADOW_MODE`).

One alternative would be to add a [runtime flag](https://github.com/salesforce/lwc/blob/849abb66745ff643de5ccb7f2a63b11f6d96acce/packages/%40lwc/features/src/flags.ts). It's a bit odd because it would never be enabled in production, but maybe it's the cleanest way.

Reply:
Runtime flags are an interesting approach which we didn't think about, but it seems that the drawback that you pointed out for the current proposal applies to both approaches.

My mental model aligns more with environment variables because, in a testing context, runtime flags feel like they are set on a per-test basis, whereas we want the entire test run to run in native mode.